lib/bloc/account/account_model.dart

```diff
@@ -20,34 +20,39 @@ enum SyncUIState {
 class AccountSettings {
   final bool ignoreWalletBalance;
   final bool showConnectProgress;
+  final bool moonpayIpCheck;
   final BugReportBehavior failePaymentBehavior;
 
   AccountSettings(
       this.ignoreWalletBalance,
       {this.showConnectProgress = false,
+      this.moonpayIpCheck = true,
       this.failePaymentBehavior = BugReportBehavior.PROMPT});
   AccountSettings.start() : this(false);
 
   AccountSettings copyWith(
       {bool ignoreWalletBalance,
       bool showConnectProgress,
+      bool moonpayIpCheck,
       BugReportBehavior failePaymentBehavior}) {
     return AccountSettings(ignoreWalletBalance ?? this.ignoreWalletBalance,
-        failePaymentBehavior:
-            failePaymentBehavior ?? this.failePaymentBehavior,
-        showConnectProgress: showConnectProgress ?? this.showConnectProgress);
+        showConnectProgress: showConnectProgress ?? this.showConnectProgress,
+        moonpayIpCheck: moonpayIpCheck ?? this.moonpayIpCheck,
+        failePaymentBehavior: failePaymentBehavior ?? this.failePaymentBehavior);
   }
 
   AccountSettings.fromJson(Map<String, dynamic> json)
       : this(json["ignoreWalletBalance"] ?? false,
-            failePaymentBehavior: BugReportBehavior.values[json["failePaymentBehavior"] ?? 0],
-            showConnectProgress: json["showConnectProgress"] ?? false);
+            showConnectProgress: json["showConnectProgress"] ?? false,
+            moonpayIpCheck: json["moonpayIpCheck"] ?? true,
+            failePaymentBehavior: BugReportBehavior.values[json["failePaymentBehavior"] ?? 0]);
 
   Map<String, dynamic> toJson() {
     return {
       "ignoreWalletBalance": ignoreWalletBalance,
+      "showConnectProgress": showConnectProgress ?? false,
+      "moonpayIpCheck": moonpayIpCheck ?? true,
```

Comment:
We don't need the null fallback here for "`moonpayIpCheck`" and "`showConnectProgress`". They will always have value.

Reply:
[Fixed](https://github.com/breez/breezmobile/pull/162/commits/9bb1eabe77c2447ccb86553ccd0231efbb76737f)